pkg/meta/redis/redis.go

```diff
@@ -750,8 +750,7 @@ func (rc *RedisDB) SetRepoReference(ctx context.Context, repo string,
 		return err
 	}
 
-	locks := []string{rc.getImageLockKey(imageMeta.Digest.String()), rc.getRepoLockKey(repo)}
-	err = rc.withRSLocks(ctx, locks, func() error {
```

Comment:
what exactly is "double locking" here?

Reply:
We take both an image lock and a repo lock at the same time for the same goroutine. Anyway, we don't need to lock at the same time.